js/lib/widget.js

```diff
@@ -137,9 +145,10 @@ export class RemoteFrameBufferModel extends DOMWidgetModel {
         }
         // Pick the oldest frame from the stack
         let frame = this.frames.shift();
+        URL.revokeObjectURL(this.img_elements?.[0]?.src);
```

Comment:
Have you double-checked that this works correctly, e.g. by debugging or with a temporary console.log? One typo in the line can cause the blobs from staying in memory, IIUC.

Reply:
I can't remember if I verified this when I submitted the PR and don't currently have the capacity to work on it further.

Reply:
I confirmed that this works.